object_manager.go

```
@@ -336,33 +338,50 @@ func (objMgr *ObjectManager) GetFixedAddress(netview string, cidr string, ipAddr
 	return &res[0], nil
 }
 
-func (objMgr *ObjectManager) GetFixedAddressFromref(ref string) (*FixedAddress, error) {
-        fixedAddr := NewFixedAddress(FixedAddress{})
-        err := objMgr.connector.GetObject(fixedAddr, ref, &fixedAddr)
-        return fixedAddr, err
+func (objMgr *ObjectManager) GetFixedAddressByRef(ref string) (*FixedAddress, error) {
+	fixedAddr := NewFixedAddress(FixedAddress{})
+	err := objMgr.connector.GetObject(fixedAddr, ref, &fixedAddr)
+	return fixedAddr, err
 }
 
 func (objMgr *ObjectManager) DeleteFixedAddress(ref string) (string, error) {
-        return objMgr.connector.DeleteObject(ref)
+	return objMgr.connector.DeleteObject(ref)
 }
 
-func (objMgr *ObjectManager) UpdateFixedAddress(fixedAddrRef string, matchclient string, macAddress string, vmID string, vmName string) (*FixedAddress, error) {
+// validate string in slice for match_client
+func validateStringinSlice(value string) bool {
+	match_client := [5]string{"MAC_ADDRESS", "CLIENT_ID", "RESERVED", "CIRCUIT_ID", "REMOTE_ID"}
 
+	for _, a := range match_client {
+		if a == value {
+			return true
+		}
+	}
+	return false
+}
+
+func (objMgr *ObjectManager) UpdateFixedAddress(fixedAddrRef string, matchClient string, macAddress string, vmID string, vmName string) (*FixedAddress, error) {
 	updateFixedAddr := NewFixedAddress(FixedAddress{Ref: fixedAddrRef})
 
 	if len(macAddress) != 0 {
 		updateFixedAddr.Mac = macAddress
 	}
 
-	if vmID != "" &&vmName != ""{
-		ea := objMgr.getBasicEA(true)
+	ea := objMgr.getBasicEA(true)
+	if vmID != "" {
 		ea["VM ID"] = vmID
+		updateFixedAddr.Ea = ea
+	}
+	if vmName != "" {
 		ea["VM Name"] = vmName
 		updateFixedAddr.Ea = ea
 	}
-
-	if matchclient != "" {
-	updateFixedAddr.MatchClient = matchclient
+	if matchClient != "" {
+		if validateStringinSlice(matchClient) {
+			updateFixedAddr.MatchClient = matchClient
+		} else {
+			fmt.Printf("wrong value for match_client passed %s \n ", matchClient)
```

why not return an error here ?